src/components/ChatEntry.js

```diff
@@ -1,22 +1,43 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
 const ChatEntry = (props) => {
+
+  const toggleLikeButton = () => {
+    props.updateLiked(props.id);
+  };
+
+  // Note: Tests for waves don't pass if I have the commented-out lines :,( 
+  // But if I don't use colorData, the tests will pass. 
+  // Just wanted to do the optional enhancement. It works in browser!
+
+  // const textColor = props.colorData[props.sender];
```

Comment:
If you want the tests to pass, you have to make sure `props.colorData` is actually optional. If this line is uncommented, it's not optional because the function will try to assume it is an object that has a certain key! Checking `colorData` it before using it will fix this:

```javascript
const textColor = props.colorData ? props.colorData[props.sender] : '';
```